bundles/admin/admin-announcements/publisher/AnnouncementsTool.js

```diff
@@ -0,0 +1,357 @@
+            const title = me.localization.tool.popup.title;
+            const content = me.templates.announcementsPopup.clone();
+
+            closeButton.setTitle(me.localization.tool.popup.close);
+            closeButton.setHandler(function () {
+                popup.close(true);
+                me.isAnnouncementsDialogOpen = false;
+            });
+
+            me.announcements.forEach(announcement => {
+                const announcementInput = me.templates.inputCheckbox.clone();
+                const annName = announcement.title;
+                const annTime = announcement.begin_date.replace(/-/g, '/') + ' - ' + announcement.end_date.replace(/-/g, '/');
+
+                announcementInput.find('input[type=checkbox]').attr({
+                    'id': announcement.id,
+                    'name': 'announcement',
+                    'value': announcement.title
+                });
+                announcementInput.find('label').html(annName).attr({
+                    'for': announcement.id
+                });
+
+                if (me.isAnnouncementValid(announcement)) {
+                    content.find('div#ann-time').append('<div>' + annTime + '</div>');
+                    if (me.shouldPreselectAnnouncement(announcement)) {
+                        announcementInput.find('input[type=checkbox]').prop('checked', true);
+                    }
+                    content.find('div#ann-title').append(announcementInput);
+                } else {
+                    content.find('div#ann-time').append('<div>' + annTime + '<div class="icon-warning-light"></div></div>');
+                    announcementInput.find('input[type=checkbox]').prop('disabled', true);
+                    content.find('div#ann-title').append(announcementInput);
+                }
+            });
+
+            // WHAT TO DO WHEN ANNOUNCEMENTS ARE SELECTED
+            content.find('input[name=announcement]').on('change', function () {
+                var announcement = me.announcements.find(item => item.title === jQuery(this).val());
+                // check if announcement is already checked, if is, add/remove accordingly
+                if (!this.checked) {
+                    me.selectedAnnouncements = me.selectedAnnouncements.filter(function (ann) {
+                        return ann.title !== announcement.title;
+                    });
+                    me.annTitles = me.annTitles.filter(function (e) { return e !== announcement.title; });
+
+                    me.getPlugin().updateAnnouncements(me.selectedAnnouncements);
+                } else {
+                    me.selectedAnnouncements.push(announcement);
+                    me.annTitles.push(announcement.title);
+
+                    me.getPlugin().updateAnnouncements(me.selectedAnnouncements);
+                }
+                jQuery('div.basic_publisher').find('input[name=publisher-announcements]').val(me.annTitles.toString());
```

Comment:
What does this line do? It updates a disabled input that is shown to the user on the published panel to show titles of the selected announcements?

Reply:
Yes, just displays the selected announcements on the tool sidebar

Reply:
Maybe add a comment above the line like "Shows user the currently selected announcement titles on the next to the tool". Could also show these with ul/li tags so it would be obvious that this isn't a functional input but an informative one.